torchbenchmark/util/backends/trt.py

```diff
@@ -31,27 +73,63 @@ def _fx2trt():
                 max_workspace_size=20 << 30,
             )
         else:
-            trt_model = compile(module=module,
-                                input=example_inputs,
-                                max_batch_size=model.batch_size,
-                                lower_precision=precision)
+            trt_model = compile(
+                module=module,
+                input=example_inputs,
+                max_batch_size=model.batch_size,
+                lower_precision=precision,
+            )
         model.set_module(trt_model)
+
     return _fx2trt, backend_args
 
+
 @create_backend
-def torch_trt(model: 'torchbenchmark.util.model.BenchmarkModel', backend_args: List[str]):
+def torch_trt(
+    model: "torchbenchmark.util.model.BenchmarkModel", backend_args: List[str]
+):
+    """Backend for Torch-TRT
+
+    Can be directly invoked from the command line, for example via:
+    python run.py resnet18 -d cuda -t eval --backend torch_trt --precision fp32 --truncate_long_and_double
+
+    Options include:
+        --truncate_long_and_double: Whether to automatically truncate long and double operations
+        --min_block_size: Minimum number of operations in an accelerated TRT block
+        --workspace_size: Size of workspace allotted to TensorRT
+        --ir: Which internal representation to use: {"ts", "dynamo_compile", "fx_ts_compat", ...}
+    """
     FP16 = True if model.dargs.precision == "fp16" else False
-    assert model.device == "cuda" and model.test == "eval", f"fx2trt only works on CUDA inference tests."
+    assert (
+        model.device == "cuda" and model.test == "eval"
+    ), f"Torch-TRT only works on CUDA inference tests."
+
+    # Extract relevant Torch-TRT arguments from the provided CLI arguments
+    torch_trt_kwargs, backend_args = parse_torch_trt_args(backend_args)
+
     def _torch_trt():
+        """Helper function for invoking Torch-TRT"""
         import torch_tensorrt
+
         module, example_inputs = model.get_module()
-        if FP16:
-            torchtrt_dtype = torch_tensorrt.dtype.half
-            torch_dtype = torch.half
-        else:
-            torchtrt_dtype = torch_tensorrt.dtype.float
-            torch_dtype = torch.float32
-        trt_input = [torch_tensorrt.Input(shape=example_inputs[0].shape, dtype=torch_dtype)]
-        trt_module = torch_tensorrt.compile(module, inputs=trt_input, enabled_precisions=torchtrt_dtype)
+        torch_dtype_precision = torch.half if FP16 else torch.float32
+
+        trt_input = [
+            torch_tensorrt.Input(shape=input_.shape, dtype=input_.dtype)
+            for input_ in example_inputs
```

Comment:
Is it necessary to add check that the type of `example_inputs` is `List[tensor]`? Actually, many models have different types of inputs.

Reply:
One suggestion is to use pytree to traverse the input and cast them to `torch_tensorrt.Input`. Similar to this: https://github.com/pytorch/benchmark/blob/9d84f9e6edcae65b43604e5781c828a05121df78/torchbenchmark/util/env_check.py#L186

Reply:
Since the different `ir` choices for Torch-TRT process inputs differently, but all can handle Torch Tensor inputs, I passed the example inputs directly to the compiler instead, so the selected `ir` can handle the inputs/casting as necessary.